src/main/java/pan/TaskList.java

```diff
@@ -0,0 +1,135 @@
+package pan;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import pan.enums.TaskStatus;
+import pan.exceptions.TaskIndexException;
+
+public class TaskList {
+    private List<Task> tasks;
+    private Storage storage;
+
+    /**
+     * Constructs a TaskList instance.
+     *
+     * @param storage Storage instance that would handle the writing of the state of taskList to a .txt file.
+     */
+    public TaskList(Storage storage) {
+        this.tasks = new ArrayList<Task>();
+        this.storage = storage;
+    }
+
+    /**
+     * Gets the tasks attribute of the TaskList instance.
+     *
+     * @return List containing Task instances that have beena added by the user.
+     */
+    public List<Task> getTasks() {
+        return tasks;
+    }
+
+    /**
+     * Adds a new task to the Task List.
+     *
+     * @param instruction New Task Instance that the user wants to add to the Task List.
+     */
+    public String add(Task instruction) {
+        tasks.add(instruction);
+        System.out.println();
+        this.storage.save(tasks);
+        return "Got it. I've added this task:\n\t" + instruction.toString() + "\nNow you have " + tasks.size() + " tasks in the list.";
+    }
+
+    /**
+     * Updates the completion status of a given task within the Task List to completed.
+     *
+     * @param index Numeric Index of the Task's position within a Task List.
+     * @throws TaskIndexException Occurs whenever the Index provided is out of bounds or do not exist.
+     */
+    public String mark(int index) throws TaskIndexException {
+        if (index > tasks.size() || index < 0) {
+            throw new TaskIndexException("You have entered an invalid index!");
+        } else {
```

Comment:
Perhaps the else block can be omitted? Since it won't be executed when an exception is thrown. I noticed this in other places as well.